### PR TITLE
Add Terraform configs for VMware vCloud Director

### LINKED ISSUE #1988: Sample terraform configs for VMware Cloud Director

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1988

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add example Terraform configs for VMware vCloud Director
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,9 +53,9 @@ care of setting up a production ready Highly Available cluster!
 ### Native Support For The Most Popular Providers
 
 KubeOne natively supports the most popular providers, including AWS, Azure,
-DigitalOcean, GCP, Hetzner Cloud, Nutanix, OpenStack, and VMware vSphere. The
-natively supported providers enjoy additional features such as integration with
-Terraform and Kubermatic machine-controller.
+DigitalOcean, GCP, Hetzner Cloud, Nutanix, OpenStack, VMware vCloud Director,
```
Comment (kron4eg) [inline README.md:56]: I think mentioning vCloud Director in the readme before providing actual integration is a bit too optimistic :)

**Review by kron4eg — COMMENTED** [review]
Small comment to address

File: examples/terraform/vcloud-director/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+The VMware vCloud Director  Quickstart Terraform configs can be used to create the needed
+infrastructure for a Kubernetes HA cluster. Check out the following
+[Creating Infrastructure guide][docs-infrastructure] to learn more about how to
+use the configs and how to provision a Kubernetes cluster using KubeOne.
+
+[docs-infrastructure]: https://docs.kubermatic.com/kubeone/master/guides/using_terraform_configs/
+
+## Setup
+
+In this setup, we assume that a dedicated org VDC has been created. It's connected to an external network using an edge gateway. Moreever, for exposing the Kubernetes API server, we need a public IP address and load balancer should be enabled at the edge gateway.
```
[the remark this answers is not in the document]
Comment (ahmedwaleedmalik) [reply examples/terraform/vcloud-director/README.md:12]: The idea is to add that documentation to `docs.kubermatic.com` and then just add the links here.

**Pull request merged**